### ISSUE TITLE
Grinds to a halt on large repos in insiders edition in v1.36

### ISSUE THREAD

**Issue by MikeRatcliffe** [mrcrowl/vscode-hg#72]
Version 1.3.1 makes VSCode Insiders Edition unusable on large repos e.g. `https://hg.mozilla.org/mozilla-central/`.

VSCode starts normally and there are no errors but the source control tab is never fully refreshed and after a few minutes everything grinds to a very slow stop. I think it is somehow filling the computer's 16 Gig of memory.

Here is the log from a working version (v1.35.0):

```
Using hg 4.9 from /usr/local/bin/hg
hg root
hg paths
hg summary -q
hg bookmarks
hg status -C
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg incoming -q
hg outgoing -q
hg cat devtools/client/inspector/inspector.js
hg summary -q
hg bookmarks
hg status -C
hg summary -q
hg bookmarks
hg status -C
hg cat devtools/client/inspector/inspector.js
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg incoming -q
hg outgoing -q
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg log -T {rev}:{node}:{date|hgdate}:{author|person}:{branch}:{join(bookmarks,'\t')}:{sub('[\n\r]+',' ',desc)}\n -r head() and not closed()
hg incoming -q
hg outgoing -q
hg summary -q
hg bookmarks
hg status -C
hg cat devtools/client/inspector/inspector.js
hg summary -q
hg bookmarks
hg status -C
hg summary -q
hg bookmarks
hg status -C
hg summary -q
hg bookmarks
hg status -C
hg summary -q
hg bookmarks
hg status -C
hg summary -q
hg bookmarks
hg status -C
```

But here is the log from Insiders edition (v1.36):

```
Using hg 4.9 from /usr/local/bin/hg
hg root
hg paths
hg root
hg paths
hg root
hg root
hg root
hg summary -q
hg paths
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg paths
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg root
hg bookmarks
hg status -C
hg paths
hg root
hg root
hg paths
hg root
hg paths
hg root
hg root
hg paths
hg root
hg paths
hg root
hg root
hg root
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg paths
hg root
hg root
hg root
hg root
hg paths
hg root
hg paths
hg root
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg root
hg root
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg root
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg bookmarks
hg status -C
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg root
hg root
hg paths
hg root
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg paths
hg root
hg root
hg root
hg paths
hg root
hg root
hg root
hg paths
hg root
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg summary -q
hg paths
hg root
```

**Comment by incidentist** [comment]
@MikeRatcliffe I accidentally triggered this bug on a large repo, but I was working on something else at the time and didn't investigate. An hour later, I tried reproducing it deliberately on the same large repo, and couldn't. I also cloned mozilla-central and could not reproduce it there (though it is slow). If you still use the extension: has this problem gone away? If not, I suspect it might go away when we release 1.6.0.

My best guess is that some cache somewhere is being generated when you first open the repo, and that slows it down enough to trigger some corner case where `hg root` gets run on an infinite loop. I did try deleting .hg/cache and still could not repro it. And this does not explain why a new mozilla-central clone didn't have the problem. But I'd like to know if you still see it.
